content/francineblanc/how-to-use-git-for-beginners.md

```diff
@@ -0,0 +1,219 @@
+
+Let's take a closer look at steps 2 and 5.
+
+### Fetching/Merging/Pulling
+To make sure the local repo is up to date with the remote, changes need to be fetched and merged from the remote and integrated into the local branch. This can be done in two stages using ```git fetch``` and ```git merge``` or in one step using ```git pull```.
+
+```git fetch``` downloads files and commits from a remote repository into the local repo. It is used to see what changes have been made to the remote before they are merged into the local repository.
+
+Fetching all branches from the remote can be achieved with ```git fetch <remote-name>```.
+
+Alternatively, a specific branch can be fetched with ```git fetch <remote-name><branch-name>```.
+
+Lastly, all registered remotes and their associated branches can be achieved with ```git fetch --all```.
+
+```git fetch``` does not automatically merge changes made in the remote repository to the local repo, so is useful in cases where code needs to be reviewed before being locally merged. In order to merge changes into the local repo, the usual ```git merge``` command is required.
+
+The ```git fetch <remote>``` and ```git merge origin <local-branch>``` command is combined as one into the ```git pull``` command, which automatically fetches and merges changes from a remote into the local branch. ```git pull``` can be run as a standalone command or with options such as ```git pull <remote-name>``` which will fetch and merge a specified remote with the local branch.
+
+It is important to form a habit of fetching and merging or pulling when working in a team to make sure the codebase being worked on locally is always up to date with the remote.
+
+### Pushing
+When collaborating on a project, it is often the case that to (amongst other reasons) prevent broken code from being added to the main development branch and/or to ensure code is correctly formatted and follows whatever code conventions have been set by the team, those who have not been working on the new feature or fix will be asked to review the code. 
+
+Code reviews are an important part of quality assurance in software development and not only helps protect against broken code from being mistakenly included in a codebase but also helps share knowledge amongst a team and can help develop the skills of both the reviewer and author.
+
+One of the advantages of creating branches, rather than working off ```main``` is that any changes made to that branch do not affect the work of any other developers in the team. When a branch is pushed up to the remote, the team members can also take the time to review the code properly without needing to worry that there may be broken code included in ```main```.
+
+To push code to the remote repo, the ```git push``` command is used as follows:
+```
+git push <name-of-remote-branch><local-branch-name>
```

Comment:
```suggestion
git push <remote-branch-name><local-branch-name>
```

Not sure but wouldn't we also need a `<remote>` as well?

```
git push <remote> <remote-branch-name>:<local-branch-name>
```

Reply:
I didn't think so @Dusch4593 ? The syntax is `git push <remote><local>` I thought? But I am prepared to be wrong :)

Reply:
Hmmm I believe I see what's being done here 😄 But I wonder if there should at least be a single space between `<remote>` and `<local>`?

Pseudo:
```
git push <remote> <local>
```

Actual:
```
git push origin my-branch
```

Reply:
@Dusch4593 yes you're absolutely correct! I have updated this now.